pkg/chunk/azure/blob_storage_client.go

```diff
@@ -77,14 +77,22 @@ func (b *BlobStorage) getChunk(ctx context.Context, decodeContext *chunk.DecodeC
 		return chunk.Chunk{}, err
 	}
 
-	buf := make([]byte, 0, b.cfg.DownloadBufferSize)
+	// Request access to the blob
+	downloadResponse, err := blockBlobURL.Download(ctx, 0, azblob.CountToEnd, azblob.BlobAccessConditions{}, false)
+	if err != nil {
+		return chunk.Chunk{}, err
+	}
+
+	// download the contents & read into a buffer
+	bodyStream := downloadResponse.Body(azblob.RetryReaderOptions{MaxRetryRequests: b.cfg.MaxRetries})
```

Comment:
I believe we need to add `defer bodyStream.Close()` right after this line, otherwise the `io.ReadCloser` is never closed. Am I missing anything?

Reply:
yes, yes we do. Thank you for spotting this